pandas/core/series.py

```diff
@@ -1195,12 +1195,14 @@ def reset_index(self, level=None, drop=False, name=None, inplace=False):
         inplace = validate_bool_kwarg(inplace, 'inplace')
         if drop:
             new_index = com._default_index(len(self))
-            if level is not None and isinstance(self.index, MultiIndex):
+            if level is not None:
                 if not isinstance(level, (tuple, list)):
                     level = [level]
+                level = set(level)
                 level = [self.index._get_level_number(lev) for lev in level]
-                if len(level) < len(self.index.levels):
-                    new_index = self.index.droplevel(level)
+                if isinstance(self.index, MultiIndex):
+                    if (len(level) < len(self.index.levels)):
```

Comment:
``len(self.index.levels)`` can be replaced with ``self.index.nlevels`` (and then I guess you'll be able to drop the ``MultiIndex`` test above). But is this test actually needed? (It's not a rhetorical question)

Reply:
Thanks for the suggestion. 

Yes replacing `len(self.index.levels)` with `self.index.nlevels` helps and `MultiIndex` test can then be removed. However, have kept the `MultiIndex` be kept to avoid any unforeseen outcomes - keeping it also produces the same outcomes (at least in the test cases checked).

`len(level) < self.index.nlevels` test is needed, in case this is dropped (assuming `MultiIndex` test is also dropped) then following errors are observed for these cases:
*Test Data*
`arrays = [np.array(['bar', 'bar', 'baz', 'baz']), np.array(['one', 'two', 'one', 'two'])]`
`s = pd.Series(range(4), name='foo', index=pd.MultiIndex.from_arrays(arrays, names=['a', 'b']))`
*Test Case*
`s.reset_index(['a', 'b'], drop=True)`
*Outcome*
`ValueError: Must pass non-zero number of levels/labels`

*Test Data*
`s = pd.Series(range(4), index=pd.RangeIndex(4, name='valid'))`
*Test Cases*
`s.reset_index(['valid', 'valid'], drop=True)`
`s.reset_index(['valid'], drop=True)`
`s.reset_index('valid', drop=True)`
*Outcome for each of 3 test cases above*
`AttributeError: 'RangeIndex' object has no attribute 'droplevel'`

Reply:
> len(level) < self.index.nlevels test is needed

Right, hadn't considered that zero-levels ``MultiIndex``es can't exist.

The ``isinstance(self.index, MultiIndex)`` doesn't make much sense but it's OK to leave it, as it guarantees that if ``pd.Series(range(4)).reset_index([], drop=True)`` doesn't fail (``pd.Index.droplevel()`` doesn't exist). Could you add a test case for this?

Reply:
done